### DISCUSSION TITLE
query_timezone parameter not taken in account in method : handle_rule_execution

### DISCUSSION THREAD

**Discussion by Majic-eyes** [jertel/elastalert2#1109] [Q&A] [answered]
Hello, i could use the 'query_timezone' parameter to have my rule planned at the correct time for my zone (UTC+1) but i'am also using the 'limit_execution' parameter and i noticed that the method using it  (handle_rule_execution in elastalert.py) does not take it in account.
I think it may cause delay in rule next run calculation.
Can you please check if it is necessary to modify method to handle 'query_timezone' ? or if there is no impact.

Regards  

**Update** : i have a rule with this configuration : 
**limit_execution**: "* 6-23 * * 1-6"
My run every, timeframe and buffer times are 5 minutes. 
i received an alert à 7:02 this morning 
I see that log : 
**elastalert Ran E-002 Detection - traffic trop bas 6h-23h from 2023-02-16 01:02 CET to 2023-02-16 07:00 CET: 6280 query hits (0 already seen), 84 matches, 1 alerts sent**
It looks like there is indeed a 1 hour decay for the limit_execution pause.

Regards

**Comment by jertel** [comment, marked as answer]
Some background: The ElastAlert2 process itself, and its internal rule scheduler, operate in UTC. This is common practice for production servers. The `query_timezone` is unrelated to scheduling, and instead relates to the data timestamps stored in Elasticsearch. I recommend ingesting Elasticsearch data in UTC. Keeping both production server time and data timestamps in UTC avoids many headaches. 

To answer your question, I have no personal need or interest in making ElastAlert2's scheduler operate on specific time zones and I don't recall many others requesting this either. If you feel that having ElastAlert2 scheduling be time zone aware is most appropriate for your use case then you can look into submitting a PR (following the contribution guidelines - see #11) to this project. But the PR must not interfere with existing functionality or otherwise cause a breaking change to existing users.

**Comment by Majic-eyes** [comment]
Hello, thank you for your answer. 
I understand how complicated it can be handling timezone for such features indeed. 
Actually i could bypass this problem using an enhancement. 

Regards
